### ISSUE TITLE
Q: How many genetic algorithms for feature selections are out there?

### ISSUE THREAD

**Issue by BradKML** [manuel-calzolari/sklearn-genetic#47]
How is this repo different from others? https://github.com/kaushalshetty/FeatureSelectionGA
P.S. are there similar algorithms to biology or ecology worth considering? https://github.com/jaswinder9051998/zoofs

**Comment by manuel-calzolari** [comment]
Hi @BradKML, see my answer here: https://github.com/manuel-calzolari/sklearn-genetic/issues/44

As for zoofs, I had never seen it before. In addition to a genetic algorithm, it seems to support other metaheuristics such as PSO. It, too, was created after my library, which I developed about 8 years ago.

**Comment by manuel-calzolari** [comment]
As for FeatureSelectionGA, that is also a library similar to mine and released after mine.

**Comment by BradKML** [comment]
To be more comprehensive, what are your thoughts in regards to advancements of genetic algorithms used in feature selection? Are there any significant modifications that are worth mentioning for the last decade?

**Comment by manuel-calzolari** [comment]
I think genetic algorithms are no longer a novelty and are fairly well established. Their application to feature selection is not rocket science, you simply apply the genetic algorithm to a binary/boolean mask that selects or does not select a given feature. It is possible to play with genetic algorithms by adding or not adding certain characteristics typical of genetic algorithms (e.g., elitism), but these are mostly techniques that have been known for many years now.

**Comment by msat59** [comment]
Nowadays, Boruta (and BorutaShap) produces more consistent results with sklearn ensemble estimators, and Baysian Optimization based feature selection (I coded for my own use) converges faster that GA and PSO at the same iteration (due having one population).

**Comment by BradKML** [comment]
@msat59 thanks for the recommendation, are there webpages that does comparison between older and SOTA algorithms regarding feature selection and/or ensemble-based predictions?

**Comment by msat59** [comment]
@BradKML , no unfortunately. I created a Notebook for myself and compared various methods. However, I used a small dataset, and the story is different with a large one. For instance, running Boruta on a dataset with 5.2k features takes about half an hour with a random forest model on a 96-core azure VM, but it takes over 8 hours with xgboost model.

If I have time (which I doubt), I will write an article about various feature selection methods.

**Comment by manuel-calzolari** [comment]
> @BradKML , no unfortunately. I created a Notebook for myself and compared various methods. However, I used a small dataset, and the story is different with a large one. For instance, running Boruta on a dataset with 5.2k features takes about half an hour with a random forest model on a 96-core azure VM, but it takes over 8 hours with xgboost model.
> 
> If I have time (which I doubt), I will write an article about various feature selection methods.

@msat59 You can also try my other library, shapicant (https://github.com/manuel-calzolari/shapicant), which generally outperforms BorutaShap. shapicant is initially faster than Boruta/BorutaShap because it does not create shadow features. However, Boruta/BorutaShap tends to become faster over time as it iteratively eliminates features, thereby reducing the number of features at runtime. In contrast, shapicant keeps the number of features constant throughout. However, the constant number of features and the independence of each iteration in shapicant allow it to be easily parallelized. For instance, if you have access to a Spark cluster, you can use shapicant’s `SparkUdfSelector` to parallelize iterations. Please note that I haven’t updated shapicant recently, so it might have compatibility issues with the latest versions of pandas or other dependencies. If you encounter any problems, feel free to open an issue, and I’ll look into it.
